packages/calypso-build/webpack/sass.js

```diff
@@ -30,12 +29,6 @@ module.exports.loader = ( { includePaths, prelude, postCssConfig = {}, cacheDire
 					},
 			  ]
 			: [] ),
-		{
-			loader: require.resolve( 'cache-loader' ),
```

Comment:
👋 This will break caching for wp-desktop. The desktop's CI config in config.yml will need to be updated (and tested) for this change as well. Desktop cache-loader saving and restoration is located [here](https://github.com/Automattic/wp-calypso/blob/7d9ea1f8227eb780c89acce1854bc3350a11fd38/.circleci/config.yml#L232-L240).

Once this logic is updated, we'd want to apply "Re-run from start" to the wp-desktop-source build to ensure that the updated cache paths are  re-built/saved and restored correctly (and that optimized build times are not affected).

Reply:
General question: why is `cache-loader` being completely dropped here? Based on recent findings, this optimization is significant.

Reply:
Follow-up: there are other parts of the CI configuration that depend on cache-loader specifically, and it's unclear how removing/replacing `cache-loader` will work. Specifically, cache-loader uses file modified time to ensure consistency between builds. [This part of the Circle config](https://github.com/Automattic/wp-calypso/blob/master/.circleci/config.yml#L686-L696) will have to be revisited as well.

Reply:
It is not removed, it has been deduplicated.

Before this change, `cache-loader` was loaded twice:

![image](https://user-images.githubusercontent.com/975703/91014348-87ff0800-e5e9-11ea-9892-85c8b42e217a.png)

The change in this PR removes the second instance. Now `cache-loader` is only added if there is a `cacheDirectory` in the options, so the consumer can specify where they want the cache and this lib doesn't have to make any assumptions as where that cache will be located.

We could add a default here, but that will remove the control from the consumer. This way (without the default), the consumer can either enable the cache and have control over the cache path, or disable the cache (by no providing a cache path).

Reply:
I updated CircleCI config to cache the whole `./cache` (not only the CSS). In theory it should be a bit faster now, I'll run the build a few times to get numbers so we can compare with master.

Reply:
With the new cache, when the cache is full it takes a bit above 8 minutes to build the source:

![image](https://user-images.githubusercontent.com/975703/91144345-30799e80-e6b4-11ea-83a9-2d81343c97c0.png)

Which seems to be at least ~4 minutes faster than master:

![image](https://user-images.githubusercontent.com/975703/91144438-5010c700-e6b4-11ea-97f9-130428589782.png)

Reply:
> I updated CircleCI config to cache the whole ./cache (not only the CSS).

This has the potential to break with TypeScript. If TypeScript finds valid cache files, it won't do any work. That means that the build output will not be produced under some circumstances if only the cache is persisted. We need to make sure that both the output and the cache are persisted, or neither.

Reply:
Oh I didn't know that :( Seems like it is not safe to persist `tsBuildInfoFile`s 

Very good catch

Reply:
An alternative I've been meaning to propose is to let TypeScript determine where to put the tsBuildInfoFiles by default. I believe it will put them next to the build output, which makes it easy to cache them all-or-nothing.

We do a lot of manual things that `tsc` has commands for. We don't need to manually clean TypeScript output, for example.

I haven't had an opportunity to apply these improvements to Calypso, but a couple of projects that use this kind of setup:
- Gutenberg: [build](https://github.com/WordPress/gutenberg/blob/a0f36df8a596dc9043cf407a0fd0fdb84490f9aa/package.json#L211) and [clean](https://github.com/WordPress/gutenberg/blob/a0f36df8a596dc9043cf407a0fd0fdb84490f9aa/package.json#L208).
- Gutenberg is quite complex, but I think it's a good model for what the Calypso tsc setup should be like.. Tannin is a smaller project that's easier to grok: https://github.com/aduth/tannin/pull/4

Reply:
Raised https://github.com/Automattic/wp-calypso/pull/45178 for removing custom path for tsBuildInfo files.